src/main/scala/mesosphere/marathon/core/task/tracker/impl/InstanceTrackerActor.scala

```diff
@@ -135,7 +144,33 @@ private[impl] class InstanceTrackerActor(
         val op = InstanceOpProcessor.Operation(deadline, sender(), instanceId, instanceUpdateOp)
         updaterRef.forward(InstanceUpdateActor.ProcessInstanceOp(op))
 
-      case msg @ InstanceTrackerActor.StateChanged(ack) =>
+      case InstanceTrackerActor.StateChanged(ack) =>
+        import context.dispatcher
+
+        ack.effect match {
+          case InstanceUpdateEffect.Update(instance, _, _) =>
+            updateRepository(() => repository.store(instance), ack)
+              .pipeTo(self)(sender())
+
+          case InstanceUpdateEffect.Expunge(instance, _) =>
+            logger.debug(s"Received expunge for ${instance.instanceId}")
+            updateRepository(() => repository.delete(instance.instanceId), ack)
+              .pipeTo(self)(sender())
+
+          case _ =>
+            Future.successful(RepositoryStateUpdated(ack)) // forward only
```

Comment:
Future into pipe? Why not simply [forward](https://doc.akka.io/docs/akka/current/actors.html?language=scala#Forward_message) the message e.g. `self forward RepositoryStateUpdated(ack)`? ;)